webapp/components/DataQuery/ButtonBar/ButtonBar.js

```diff
@@ -0,0 +1,73 @@
+import './buttonBar.scss'
+import React from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames'
+
+import { Query } from '@common/model/query'
+
+import { useIsAppSaving } from '@webapp/store/app'
+import { useAuthCanCleanseRecords } from '@webapp/store/user'
+
+import ButtonDownload from './ButtonDownload'
+import ButtonFilter from './ButtonFilter'
+import ButtonSort from './ButtonSort'
+
+const ButtonBar = (props) => {
+  const { nodeDefsSelectorVisible, query, onChangeQuery, setNodeDefsSelectorVisible } = props
+
+  const appSaving = useIsAppSaving()
+  const canEdit = useAuthCanCleanseRecords()
+  const modeEdit = Query.isModeRawEdit(query)
+  const modeAggregate = Query.isModeAggregate(query)
+  const hasSelection = Query.hasSelection(query)
+  const hasData = false // TODO
+
+  return (
+    <div className="data-query-button-bar">
+      <div>
+        <button
+          type="button"
+          className={classNames('btn', 'btn-s', { highlight: nodeDefsSelectorVisible })}
+          onClick={() => setNodeDefsSelectorVisible(!nodeDefsSelectorVisible)}
+        >
+          <span className="icon icon-tab icon-14px" />
+        </button>
+        <button
+          type="button"
+          className={classNames('btn', 'btn-s', 'btn-edit', { highlight: Query.isModeAggregate(query) })}
+          onClick={() => onChangeQuery(Query.toggleModeAggregate(query))}
+          aria-disabled={appSaving || modeEdit || !nodeDefsSelectorVisible}
+        >
+          <span className="icon icon-sigma icon-14px" />
+        </button>
+        {canEdit && hasSelection && (
+          <button
+            type="button"
+            className={`btn btn-s btn-edit${modeEdit ? ' highlight' : ''}`}
```

Comment:
shall we use this **classNames**?
I personally don't like it because it is not plain html/jsx but it is like when we extract the className calculation.

Reply:
I will use classNames in next PR. why don't you like it? makes the code simpler